src/Arcus.Scripting.KeyVault/Scripts/Get-AzKeyVaultAccessPolicies.ps1

```diff
@@ -0,0 +1,57 @@
+param(
+   [string][parameter(Mandatory = $true)] $keyVaultName,
+   [string][parameter(Mandatory = $false)] $resourceGroupName = ""
+)
+
+$keyVault = $null
+if($resourceGroupName -eq '')
+{
+	Write-Host "Looking for the Key Vault with name '$keyVaultName'."
+	$keyVault = Get-AzKeyVault -VaultName $keyVaultName
+}
+else
+{
+	Write-Host "Looking for the Key Vault with name '$keyVaultName' in resourcegroup '$resourceGroupName'"
+	$keyVault = Get-AzKeyVault -VaultName $keyVaultName -ResourceGroupName $resourceGroupName
+}
+
+if($keyVault)
+{
+	$armAccessPolicies = @()
+	$keyVaultAccessPolicies = $keyVault.accessPolicies
+
+	if($keyVaultAccessPolicies)
+	{
+	   Write-Host "Key Vault '$keyVaultName' is found."
+
+	   foreach($keyVaultAccessPolicy in $keyVaultAccessPolicies)
+	   {
+		  $armAccessPolicy = [pscustomobject]@{
+			 tenantId = $keyVaultAccessPolicy.TenantId
+			 objectId = $keyVaultAccessPolicy.ObjectId
+		  }
+
+		  $armAccessPolicyPermissions = [pscustomobject]@{
+			 keys =  $keyVaultAccessPolicy.PermissionsToKeys
+			 secrets = $keyVaultAccessPolicy.PermissionsToSecrets
+			 certificates = $keyVaultAccessPolicy.PermissionsToCertificates
+			 storage = $keyVaultAccessPolicy.PermissionsToStorage
+		  }
+
+		  $armAccessPolicy | Add-Member -MemberType NoteProperty -Name permissions -Value $armAccessPolicyPermissions
+
+		  $armAccessPolicies += $armAccessPolicy
+	   }   
+	}
+
+	$armAccessPoliciesParameter = [pscustomobject]@{
+		list = $armAccessPolicies
+	}
+
+	Write-Host "Current access policies: $armAccessPoliciesParameter"
+	return $armAccessPoliciesParameter
+}
+else
+{
+	Write-Host "KeyVault '$keyVaultName' could not be found."
```

Comment:
Write-Host or Write-Error?

Reply:
Could go either way:
- On the initial execution of a pipeline, the KV doesn't exist yet, so you cannot retrieve the access policies, but you don't want the command to fail in this case.
- On the other side, trying to retrieve access policies of a non-existing KV is correct to fail..

Perhaps a better option would be:
- add a switch-parameter to indicate whether it should `Write-Host` or `Write-Error` (default = `Write-Error`)

Reply:
Hmm, shouldn't that be the job of the common parameter, `-ErrorAction`? Couldn't we use always `WriteError` and let the consumer decide with `ErrorAction` and `ErrorVariable` how errors should be processed?

Reply:
You are right @mbraekman, didn't think of the first scenario; in that case I would use `Write-Warning` to be explicit but that it can be ok - Thoughts?

Reply:
Let's go for `Write-Warning` by default and let the consumer decide otherwise using `-ErrorAction`, as suggested by Stijn.

Reply:
Ok, changed that.